crates/bevy_core/src/bytes.rs

```diff
@@ -50,9 +50,8 @@ where
 {
     fn from_bytes(bytes: &[u8]) -> Self {
         unsafe {
-            let byte_ptr = bytes.as_ptr();
-            let ptr = byte_ptr as *const Self;
-            (*ptr).clone()
+            let ptr = bytes.as_ptr() as *const Self;
+            std::ptr::read_unaligned(ptr)
```

Comment:
Now that I think about it. I still don't think this is right. 
This only works if `T` is `Copy`, however, `T` is only bounded on `Clone` :/

Reply:
So perhaps something like:
``rust
fn from_bytes(bytes: &[u8]) -> Self {
    let value = unsafe { bytes.as_ptr().cast::<Self>().read_unaligned() };
    let clone = value.clone();
    std::mem::forget(value);
    clone 
}
```

would work?